sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filtering/DependencyDataColumns.java

```diff
@@ -0,0 +1,95 @@
+package com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.models.RemoteDependencyData;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.utils.FormattedDuration;
+
+// Casing of private fields is to match the names of fields passed down via filtering configuration
+public class DependencyDataColumns extends TelemetryColumns {
+    private String Target;
```

Comment:
In Java, the convention is that the fields start with a lower case.

Reply:
I'm aware - I mainly set the casing this way because reflection is used to fetch the names of the filters in checkFilter() based on a string that is passed to us from the response body of the /post call.

Do you prefer that reflection not be used, or that I create a map of field names that are passed to us vs the field name in the object?

Reply:
I would prefer to avoid reflection. It's a bit surprising to have the code broken if we rename a property. 😅

Reply:
Ok I will rethink this design a bit & update

Reply:
I'm going to try out this idea:
TelemetryColumns will be an abstract class where I add another abstract method called getFieldValue(String fieldName). This method will be implemented differently in each of the subclasses & there won't be a dependency on the casing of the fields or reflection. 

This does mean that I will keep the inheritance structure that you suggested removing earlier in favor of composition; but considering the new abstract method added, I think it should be fine?

Reply:
See https://github.com/Azure/azure-sdk-for-java/pull/43040#discussion_r1856923182

I would prefer an interface defining the following contract: provide the value of a column following a column type. The custom properties code would not be a part of this interface (contract) and be used in the implementations of this interface / contract.